java/src/main/java/com/google/appengine/tools/pipeline/impl/model/JobInstanceRecord.java

```diff
@@ -44,11 +45,11 @@
   // transient
   private Job<?> jobInstance;
 
-  public JobInstanceRecord(JobRecord job, Job<?> jobInstance) {
+  public JobInstanceRecord(JobRecord job, Job<?> jobInstance, JobSetting.DisplayName displayNameSetting) {
     super(job.getRootJobKey(), job.getGeneratorJobKey(), job.getGraphGuid());
     jobKey = job.getKey();
     jobClassName = jobInstance.getClass().getName();
-    jobDisplayName = jobInstance.getJobDisplayName();
+    jobDisplayName = (displayNameSetting == null ? jobInstance.getJobDisplayName() : displayNameSetting.getValue());
```

Comment:
Ah, bummer... I kind of like the fact that job settings take precedence over everything else. I get what you are saying though. Will fix up some changes and re-submit.